lightning/src/ln/channelmanager.rs

```diff
@@ -8973,7 +8973,7 @@ where
 	/// # Limitations
 	///
 	/// Requires a direct connection to an introduction node in [`Offer::paths`] or to
-	/// [`Offer::signing_pubkey`], if empty. A similar restriction applies to the responding
+	/// [`Offer::issuer_id`], if empty. A similar restriction applies to the responding
```

Comment:
Don't we have the same problem with `payer_id`?

Reply:
Hmm... I thought the reason the spec was changed from `node_id` to `issuer_id` was because `node_id` implied it was some persistent id. Maybe `issuer_signing_pubkey` and `payer_signing_pubkey`?

Reply:
> Or `invoice_signing_pubkey` and `invreq_signing_pubkey` to indicate that these are keys specific to this inv/req, and not specific to the issuer/payer.

I'm not so keen on those. `InvoiceRequest::invreq_signing_pubkey` is redundant and `Offer::invoice_signing_pubkey` would make it such that we have both `Bolt12Invoice::invoice_signing_pubkey` -- also redundant and unclear where it is coming from -- and `Bolt12Invoice::signing_pubkey`.

Using "issuer" and "payer" is clearer, IMO. It's not that the signing pubkey is specific to the issuer/payer but rather belongs to them.

Or thought of another way, it *is* specific to the issuer/payer and thus it actually *is* an id. It's just that they can have multiple identities -- and thus ids -- that cannot be correlated. But happy to use `issuer_signing_pubkey` and `payer_signing_pubkey` if you prefer to not use "id".

Reply:
PTAL, the latest push renamed both `issuer_id` and `payer_id` in the public API.